libs/gpio/mock/gpio.c

```diff
@@ -0,0 +1,36 @@
+#include "libs/gpio/api.h"
```

Comment:
What's the point of this if we don't have tests yet? Are tests still a WIP? Do you want to include them in this PR or nah (I don't care either way as long as you're confident this will work)?